pkg/framework/context/context.go

```diff
@@ -24,7 +24,7 @@ import (
 // Provider supplies the framework configuration to client objects.
 type Provider struct {
 	services                 []dispatcher.Service
-	messageServices          []dispatcher.MessageService
+	msgHandler               dispatcher.MessageHandler
```

Comment:
This should not be a MessageHandler interface. The framework packages will not call Register and Unregister.

Reply:
changed to `MessageServiceProvider`. Hiding Register/Unregister from framework.